gateway-framework/src/network/exchange_rate.rs

```diff
@@ -57,6 +57,8 @@ pub async fn grt_per_usd(provider: Url) -> anyhow::Result<watch::Receiver<NotNan
             if let Err(grt_per_usd_send_err) = tx.send(grt_per_usd) {
                 tracing::error!(%grt_per_usd_send_err);
             }
+
+            interval.tick().await;
```

Comment:
From tokio [documentation](https://docs.rs/tokio/latest/tokio/time/fn.interval.html):

> The first tick completes immediately.

In that case, `interval.tick().await` should be the first statement of the loop—otherwise, the first time, the task is executed twice with no spacing between executions.

Reply:
Ah, I made the false assumption that interval doesn't fire immediately :facepalm:

Reply:
It is also important to note that using the default interval behaviour for the missed ticks, the [Burst](https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html#variant.Burst) behaviour, may cause undesired non-spatiated executions:

```
interval.set_missed_tick_behavior(MissedTickBehavior::Burst);


Expected ticks: |     1     |     2     |     3     |     4     |     5     |     6     |
Actual ticks:   | work -----|          delay          | work | work | work -| work -----|
```

I have not checked wich is the right behaviour, but it should be one of these two: [`Delay`](https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html#variant.Delay) or [`Skip`](https://docs.rs/tokio/latest/tokio/time/enum.MissedTickBehavior.html#variant.Skip).

Reply:
> Ah, I made the false assumption that interval doesn't fire immediately 🤦

It needs to know the time origin; that's the reason for the 1st tick to complete immediately.